src/main/java/user_use_case/interactors/UpgradeUserInteractor.java

```diff
@@ -0,0 +1,35 @@
+package user_use_case.interactors;
+
+import entities.OwnerFactory;
+import entities.OwnerUser;
+import user_use_case.gateways.UserGateway;
+
+import java.util.ArrayList;
+
+public class UpgradeUserInteractor implements UpgradeUserInputBoundary{
+    private final OwnerFactory factory;
+
+    private final UserGateway gateway;
+
+    public UpgradeUserInteractor(OwnerFactory factory, UserGateway gateway) {
+        this.factory = factory;
+        this.gateway = gateway;
+    }
+
+    @Override
+    public UpgradeUserResponseModel UpgradeUser(UpgradeUserRequestModel requestModel) {
+        String username = requestModel.getUsername();
+        String password = requestModel.getPassword();
+        ArrayList<String> reviews = requestModel.getPastReviews();
+        ArrayList<String> liked_reviews = requestModel.getLikedReviews();
+        int received_reports = requestModel.getReceivedReports();
+        boolean banned = requestModel.isBanned();
+        ArrayList<String> ownedRestaurants = new ArrayList<>();
+
```

Comment:
it would be easier to pass through user in the request model, that already contains all the info u need